rpc/src/rpc.rs

```diff
@@ -166,6 +182,20 @@ impl JsonRpcConfig {
             ..Self::default()
         }
     }
+    pub fn default_for_test_with_secondary_indexes() -> Self {
+        Self {
+            full_api: true,
+            account_indexes: solana_runtime::accounts_index::AccountSecondaryIndexes {
+                keys: None,
+                indexes: HashSet::from([
+                    solana_runtime::accounts_index::AccountIndex::ProgramId,
+                    solana_runtime::accounts_index::AccountIndex::SplTokenMint,
+                    solana_runtime::accounts_index::AccountIndex::SplTokenOwner,
+                ]),
+            },
+            ..Self::default()
+        }
+    }
```

Comment:
Switching the test-validator parameter should lead to removing this method.

Reply:
Possibly yes, that was actually exactly what I was working on when I had to bounce for travel, mimicking the regular validator args have me initializing a bit differently because now its not a fixed struct and is passed in and generated when parsing the args.